Imagine/Filter/Loader/ScaleFilterLoader.php

```diff
@@ -70,7 +70,11 @@ public function load(ImageInterface $image, array $options = array())
             if (null == $width || null == $height) {
                 $ratio = max($widthRatio, $heightRatio);
             } else {
-                $ratio = min($widthRatio, $heightRatio);
+                if('min' === $this->dimensionKey){
```

Comment:
Perhaps `$ratio = ('min' === $this->dimensionKey) ? max($widthRatio, $heightRatio) : min($widthRatio, $heightRatio)`?